unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Utils/ECSComponentsUtils.cs

```diff
@@ -0,0 +1,107 @@
+        
+        meshFilter.sharedMesh = mesh;
+
+        // We should remove this relation in the future, the entity shouldn't know about the mesh
+        entity.meshesInfo = meshesInfo;
+        
+        UpdateRenderer(entity,gameObject, renderers, visible, withCollisions, isPointerBlocker);
+        meshesInfo.UpdateRenderersCollection();
+        return meshesInfo;
+    }
+    
+    public static void UpdateRenderer(IDCLEntity entity,GameObject meshGameObject, Renderer[] renderers,bool visible, bool withCollisions, bool isPointerBlocker)
+    {
+        ConfigureVisibility(meshGameObject, visible,renderers);
+        
+        CollidersManager.i.ConfigureColliders(meshGameObject, withCollisions, false, entity, CalculateCollidersLayer(withCollisions,isPointerBlocker));
+    }
+    
+    public static void ConfigureVisibility(GameObject meshGameObject, bool shouldBeVisible, Renderer[] meshRenderers = null)
+    {
+        if (meshGameObject == null)
+            return;
+
+        if (!shouldBeVisible)
+        {
+            MaterialTransitionController[] materialTransitionControllers = meshGameObject.GetComponentsInChildren<MaterialTransitionController>();
+
+            for (var i = 0; i < materialTransitionControllers.Length; i++)
+            {
+                Object.Destroy(materialTransitionControllers[i]);
+            }
+        }
+
+        if (meshRenderers == null)
+            meshRenderers = meshGameObject.GetComponentsInChildren<Renderer>(true);
+
+        Collider onPointerEventCollider;
+
+        for (var i = 0; i < meshRenderers.Length; i++)
+        {
+            meshRenderers[i].enabled = shouldBeVisible;
+
+            if (meshRenderers[i].transform.childCount > 0)
+            {
+                onPointerEventCollider = meshRenderers[i].transform.GetChild(0).GetComponent<Collider>();
+
+                if (onPointerEventCollider != null && onPointerEventCollider.gameObject.layer == PhysicsLayers.onPointerEventLayer)
+                    onPointerEventCollider.enabled = shouldBeVisible;
+            }
+        }
+    }
+    
+    public static void RemoveRendereableFromDataStore(string sceneId, Rendereable rendereable)
+    {
+        DataStore.i.sceneWorldObjects.RemoveRendereable(sceneId, rendereable);
+    }
+
+    public static Rendereable AddRendereableToDataStore(string sceneId, int componentId, Mesh mesh, GameObject gameObject)
+    {
+        int triangleCount = mesh.triangles.Length;
+
+        var newRendereable =
+            new Rendereable()
+            {
+                container = gameObject,
+                totalTriangleCount = triangleCount,
+                meshes = new HashSet<Mesh>() { mesh },
+                meshToTriangleCount = new Dictionary<Mesh, int>() { { mesh, triangleCount } }
+            };
+
+        newRendereable.renderers = MeshesInfoUtils.ExtractUniqueRenderers(gameObject);
+        newRendereable.ownerId = componentId;
```

Comment:
No, because in the future the entity shouldn't be tied to a mesh, the entity could have several shapes and it should work, the renderable is attached to a component, not an entity

Reply:
That doesn't matter at the moment. That is used for scene metrics and it looks like it's expecting an entityId right now so we should put an entityId there.
When the time comes to decouple meshes from entities then we see if it is necessary to modify that

Reply:
Changed